test/parallel_api/algorithm/alg.nonmodifying/transform_if.pass.cpp

```diff
@@ -0,0 +1,83 @@
+// -*- C++ -*-
+//===-- transform_if.pass.cpp ----------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+// Tests for transform_if
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+
+#include "support/utils.h"
+#include <random>
+
+template <typename Policy, typename Queue>
+void test_transform_if(Policy policy, int input_size, Queue q) {
+    auto sycl_deleter = [q](int* mem) { sycl::free(mem, q.get_context()); };
+
+    ::std::unique_ptr<int, decltype(sycl_deleter)>
+
+    data((int*)sycl::malloc_shared(sizeof(int) * input_size, q.get_device(), q.get_context()), sycl_deleter),
```

Comment:
Here you test only USM. It's better to test `sycl::buffer` and `std::vector` iterators as well. You can take a look at other tests and/or at #247 for reference